src/browser/fetch/request.rs

```diff
@@ -0,0 +1,282 @@
+}
+
+impl<'a> Request<'a> {
+    /// Create new request based on the provided url.
+    ///
+    /// To get a [`Response`](./struct.Response.html) you need to pass
+    /// `Request` to the [`fetch`](./fn.fetch.html) function.
+    ///
+    /// [MDN reference](https://developer.mozilla.org/en-US/docs/Web/API/Request)
+    pub fn new(url: impl Into<Cow<'a, str>>) -> Self {
+        Self {
+            url: url.into(),
+            ..Self::default()
+        }
+    }
+
+    // TODO: remove when https://github.com/rust-lang/rust-clippy/issues/4979 will be fixed
+    #[allow(clippy::missing_const_for_fn)]
+    /// Set headers for this request.
+    /// It will replace any existing headers.
+    pub fn headers(mut self, headers: Headers<'a>) -> Self {
+        self.headers = headers;
+        self
+    }
+
+    /// Set specific header.
+    pub fn header(mut self, header: Header<'a>) -> Self {
+        self.headers.set(header);
+        self
+    }
+
+    /// Set HTTP method. Default method is `GET`.
+    pub const fn method(mut self, method: Method) -> Self {
+        self.method = method;
+        self
+    }
+
+    /// Set request body to provided `JsValue`. Consider using `json` or `text` methods instead.
+    ///
+    /// Note that a request using the GET or HEAD method cannot have a body.
+    pub fn body(mut self, body: JsValue) -> Self {
+        self.body = Some(body);
+
+        #[cfg(debug_assertions)]
+        match self.method {
+            Method::Get | Method::Head => {
+                error!("GET and HEAD requests shoudn't have a body");
+            }
+            _ => {}
+        }
+
+        self
+    }
+
+    /// Set request body by JSON encoding provided data.
+    /// It will also set `Content-Type` header to `application/json; charset=utf-8`.
+    ///
+    /// # Errors
+    ///
+    /// This method can fail if JSON serialization fail. It will then
+    /// return `FetchError::SerdeError`.
+    pub fn json<T: Serialize + ?Sized>(mut self, data: &T) -> Result<Self> {
+        self.headers
+            .set(header::content_type("application/json; charset=utf-8"));
+        let body = serde_json::to_string(data).map_err(FetchError::SerdeError)?;
+        self.body = Some(body.into());
+        Ok(self)
+    }
+
+    /// Set request body to a provided string.
+    /// It will also set `Content-Type` header to `text/plain; charset=utf-8`.
+    pub fn text(mut self, text: impl AsRef<str>) -> Self {
+        self.headers
+            .set(header::content_type("text/plain; charset=utf-8"));
+        self.body = Some(JsValue::from(text.as_ref()));
+        self
+    }
+
+    /// Set request mode.
+    /// It can either be `cors`, `no-cors`, `same-origin`, or `navigate`.
+    /// The default is `cors`.
+    pub const fn mode(mut self, mode: web_sys::RequestMode) -> Self {
+        self.mode = Some(mode);
+        self
+    }
+
+    /// Set request credentials.
+    /// It can either be `omit`, `same-origin`, or `include`.
```

Comment:
Could you add `# Example` block?
(The same for other methods, where it's not clear how to use enum of values from the description)

Reply:
These methods expose websys, thus they are documented on MDN/websys. Not sure what is the best way to solve this.

Reply:
What about:
```rust
/// Set request credentials.
///
/// Variants: `Omit`, `SameOrigin`, `Include`
/// Default: `SameOrigin`
///
/// [MDN reference](https://developer.mozilla.org/en-US/docs/Web/API/Request/credentials)
/// [`web_sys` reference](https://rustwasm.github.io/wasm-bindgen/api/web_sys/enum.RequestCredentials.html)
```